app/src/views/AccountMyFormsDref/DownloadImportTemplateButton/DownloadImportTemplateModal/index.tsx

```diff
@@ -0,0 +1,385 @@
+
+    await workbook.xlsx.writeBuffer().then(
+        (sheet) => {
+            FileSaver.saveAs(
+                new Blob([sheet], { type: 'application/vnd.ms-excel;charset=utf-8' }),
+                templateFileName.replace(' ', '_'),
+            );
+        },
+    );
+
+    callback();
+}
+
+interface Props {
+    onComplete: () => void;
+}
+
+function DownloadImportTemplateModal(props: Props) {
+    const { onComplete } = props;
+
+    const { dref_dref_dref_type } = useGlobalEnums();
+    const strings = useTranslation(i18n);
+
+    const [generationPending, setGenerationPending] = useState(false);
+    const [typeOfDref, setTypeOfDref] = useState<TypeOfDrefEnum>(DREF_TYPE_RESPONSE);
+
+    const { drefFormSchema, optionsMap } = useImportTemplateSchema();
+    const templateActions = createImportTemplate(drefFormSchema, optionsMap);
+
+    const drefTypeLabelMap = useMemo(
+        () => (
+            listToMap<TypeOfDrefEnum, string, string>(
+                dref_dref_dref_type,
+                (option) => option.key,
+                (option) => option.label,
+            )
+        ),
+        [dref_dref_dref_type],
+    );
+
+    const handleDownloadClick = useCallback(() => {
+        if (isNotDefined(templateActions)) {
+            return;
+        }
+
+        setGenerationPending((alreadyGenerating) => {
+            if (!alreadyGenerating) {
+                generateTemplate(
+                    templateActions,
+                    optionsMap,
+                    drefTypeLabelMap,
+                    typeOfDref,
+                    () => {
+                        setGenerationPending(false);
+                        onComplete();
+                    },
+                );
+            }
+
+            return true;
+        });
+    }, [
+        templateActions,
+        optionsMap,
+        onComplete,
+        drefTypeLabelMap,
+        typeOfDref,
+    ]);
+
+    return (
+        <Modal
+            heading={strings.heading}
+            footerActions={(
+                <Button
+                    name={undefined}
+                    onClick={handleDownloadClick}
+                    disabled={generationPending || isNotDefined(drefFormSchema)}
+                >
+                    {strings.downloadButtonLabel}
+                </Button>
+            )}
+            contentViewType="vertical"
+            spacing="comfortable"
+            onClose={onComplete}
+        >
+            <RadioInput
+                name={undefined}
+                label="Select type of DREF for template"
+                options={dref_dref_dref_type}
+                keySelector={typeOfDrefKeySelector}
+                labelSelector={stringValueSelector}
+                value={typeOfDref}
+                onChange={setTypeOfDref}
+                // Only response type is available for now
+                disabled
+            />
```

Comment:
We should disable everything except the "Response" type

Reply:
It's a requirement from client.

Reply:
Yes. Currently everything is disabled. But, we should instead disable other options.